### PR TITLE
cmake format

### PR DESCRIPTION
**Pull request by szadam**
### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

- [x] CI workflows, not executed per PR (e.g. Nightly), execute properly 
- [x] New tests added, especially if they will fail without my changes
- [x] Extended the README/documentation
- [x] All newly added source files have a license
- [x] All newly added source files are referenced in CMake files

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -156,71 +164,157 @@ if(UMF_BUILD_BENCHMARKS)
...
-    else()
-        message(FATAL_ERROR "UMF_FORMAT_CODE_STYLE=ON, but clang-format not found (required version: ${CLANG_FORMAT_REQUIRED})")
+    endif()
+
+    if(CMAKE_FORMAT)
+        get_program_version_major_minor(${CMAKE_FORMAT} CMAKE_FORMAT_VERSION)
+        message(STATUS "Found cmake-format: ${CMAKE_FORMAT} "
+                       "(version: ${CMAKE_FORMAT_VERSION})")
+
+        # Check if cmake-format (in correct version) is available for code
```
Comment (PatKamin) [inline CMakeLists.txt:200]: cmake files formatting

File: CMakeLists.txt [hunk 2]
```diff
@@ -156,71 +164,157 @@ if(UMF_BUILD_BENCHMARKS)
...
+            message(FATAL_ERROR "Required cmake-format version is"
+                                "${CMAKE_FORMAT_REQUIRED}")
+        endif()
+    endif()
+
+    if(NOT CLANG_FORMAT AND NOT CMAKE_FORMAT)
+        message(
+            FATAL_ERROR
+                "UMF_FORMAT_CODE_STYLE=ON, but neither clang-format (required version: "
+                "${CLANG_FORMAT_REQUIRED}) nor cmake-format was found.")
```
Comment (PatKamin) [inline CMakeLists.txt:213]: Print the required version of cmake-format as well.

File: CMakeLists.txt [hunk 3]
```diff
@@ -156,71 +164,157 @@ if(UMF_BUILD_BENCHMARKS)
...
             "${DIR}/**/*.hpp"
             "${DIR}/**/*.c"
-            "${DIR}/**/*.cpp"
-        )
+            "${DIR}/**/*.cpp")
     endforeach()
     file(GLOB_RECURSE format_list ${format_glob})
 
-    message(STATUS "Adding clang-format-check and clang-format-apply make targets")
+    message(STATUS "Adding clang-format-check and clang-format-apply make "
```
Comment (PatKamin) [inline CMakeLists.txt:239]: There was a discussion about this in a previous PR, but I want to make sure we agree on this: these targets are not make-specific so better to describe them as CMake targets? @lukaszstolarczuk

File: CMakeLists.txt [hunk 4]
```diff
@@ -156,71 +164,157 @@ if(UMF_BUILD_BENCHMARKS)
...
+
+    add_custom_target(
+        cmake-format-apply
+        COMMAND ${CMAKE_FORMAT} --in-place ${format_cmake_list}
+        COMMENT "Format files using cmake-format")
+
+    # Add a convenience target for running both clang-format and cmake-format
+    # checks/apply
+    add_custom_target(
+        format-checks
```
Comment (PatKamin) [inline CMakeLists.txt:295]: nit: `format-check`? to align with a singular form of the `format-apply` target

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -62,32 +62,40 @@ To enable additional checks (including `-Werror` / `/WX` compilation flag), swit
...
 
-**NOTE**: We use specific clang-format version - **15.0** is required. It can be installed,
-e.g., with command: `python -m pip install clang-format==15.0.7`.
+If you wish to use only `clang-format` or only `cmake-format`, you can execute the corresponding
+`clang-format-check` and `clang-format-apply` for source files, or `cmake-format-check` and
+`cmake-format-apply` for CMake files, respectively.
+
+**NOTE**: We use specific versions of formatting tools to ensure consistency across the project. The required versions are:
+- clang-format version **15.0**, which can be installed with the command: `python -m pip install clang-format==15.0.7`.
+- cmake-format version **0.6.13**, which can be installed with the command: `python -m pip install cmake-format==0.6`.
```
Comment (PatKamin) [inline CONTRIBUTING.md:96]: `0.6` -> `0.6.13`

File: CMakeLists.txt (hunk 3 above)
Comment (PatKamin) [reply CMakeLists.txt:239]: Yes, that's ok, you can say that it's a make target. This tool is most probably not available on Windows anyway...

File: CMakeLists.txt (hunk 1 above)
Comment (szadam) [reply CMakeLists.txt]: Done

File: README.md [hunk 6]
```diff
@@ -101,7 +102,7 @@ List of options provided by CMake:
 | UMF_BUILD_EXAMPLES | Build UMF examples | ON/OFF | ON |
 | UMF_ENABLE_POOL_TRACKING | Build UMF with pool tracking | ON/OFF | ON |
 | UMF_DEVELOPER_MODE | Treat warnings as errors and enables additional checks | ON/OFF | OFF |
-| UMF_FORMAT_CODE_STYLE | Add clang-format-check and clang-format-apply targets to make | ON/OFF | OFF |
+| UMF_FORMAT_CODE_STYLE | Add clang and cmake  -format-check and -format-apply targets to make | ON/OFF | OFF |
```
Comment (lukaszstolarczuk) [inline README.md:105]: redudant, trailing space at the end

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -62,32 +62,40 @@ To enable additional checks (including `-Werror` / `/WX` compilation flag), swit
...
 
-**NOTE**: We use specific clang-format version - **15.0** is required. It can be installed,
-e.g., with command: `python -m pip install clang-format==15.0.7`.
+If you wish to use only `clang-format` or only `cmake-format`, you can execute the corresponding
+`clang-format-check` and `clang-format-apply` for source files, or `cmake-format-check` and
+`cmake-format-apply` for CMake files, respectively.
+
+**NOTE**: We use specific versions of formatting tools to ensure consistency across the project. The required versions are:
+- clang-format version **15.0**, which can be installed with the command: `python -m pip install clang-format==15.0.7`.
+- cmake-format version **0.6.13**, which can be installed with the command: `python -m pip install cmake-format==0.6.13`.
```
Comment (lukaszstolarczuk) [inline CONTRIBUTING.md:96]: I believe you can just write `- cmake-format version **0.6** ...` patch version should not break compatibility (I don't talk about installation part - it should stay with patch version)

File: CMakeLists.txt [hunk 8]
```diff
@@ -156,71 +164,158 @@ if(UMF_BUILD_BENCHMARKS)
...
+                "${CMAKE_FORMAT_VERSION}) was found.")
     endif()
 
     # Obtain files for clang-format check
     set(format_glob)
-    foreach(DIR IN ITEMS include src test benchmark)
-        list(APPEND format_glob
+    foreach(
+        DIR IN
+        ITEMS include
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:221]: pls sort these dirs A-Z

File: CMakeLists.txt [hunk 9]
```diff
@@ -156,71 +164,158 @@ if(UMF_BUILD_BENCHMARKS)
...
+            "${DIR}/CMakeLists.txt"
+            "${DIR}/*.cmake"
+            "${DIR}/**/CMakeLists.txt"
+            "${DIR}/**/*.cmake")
+    endforeach()
+    file(GLOB_RECURSE format_cmake_list ${format_cmake_glob})
+    list(APPEND format_cmake_list "${PROJECT_SOURCE_DIR}/CMakeLists.txt")
+
+    # Print the list of files targeted for formatting by cmake-format
+    message(STATUS "Files to be formatted with cmake-format:")
```
Comment (lukaszstolarczuk) [inline CMakeLists.txt:275]: I believe you can delete it now.

IMHO, if this is to be printed always, it should be done in build time (within `make`), not as part of CMake configuration.

File: CMakeLists.txt (hunk 8 above)
Comment (szadam) [reply CMakeLists.txt:221]: Done

File: README.md (hunk 6 above)
Comment (szadam) [reply README.md]: Done

File: CONTRIBUTING.md (hunk 7 above)
Comment (szadam) [reply CONTRIBUTING.md]: Done

File: CMakeLists.txt (hunk 9 above)
Comment (szadam) [reply CMakeLists.txt]: I just deleted it here. We could add print in make for cmake-format and clang-format in other PR

File: CMakeLists.txt (hunk 8 above)
Comment (lukaszstolarczuk) [reply CMakeLists.txt:221]: not done - include after exampels

**Review by lukaszstolarczuk — COMMENTED** [review]
pls rebase + 1 minor issue and we're good to go 😉

File: CMakeLists.txt (hunk 8 above)
Comment (szadam) [reply CMakeLists.txt]: sorry, now it is good

**Review by lukaszstolarczuk — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**